db.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+import json
+import sys
+
+
+def validate(input):
+    normalized = normalize_internal(input)
+    if normalized != input:
+        sys.stderr.write("Input is not normalized.\n")
+        sys.exit(0)
+
+def normalize_internal(input):
+    db = json.loads(input)
+    string = json.dumps(normalize_db(db), ensure_ascii=False, allow_nan=False, sort_keys=False, indent=2)
+    return string + "\n"
+
+def normalize(input):
+    open("db.json", "w").write(normalize_internal(input))
+
+def normalize_db(db):
+    i = 0
+    for workstream in db["workstreams"]:
+        db["workstreams"][i] = normalize_workstream(workstream)
+        i += 1
+
+    i = 0
+    for idea in db["ideas"]:
+        db["ideas"][i] = normalize_workstream_standard_or_idea(idea)
+        i += 1
+
+    i = 0
+    for reference in db["biblio"]:
+        db["biblio"][i] = normalize_reference(reference)
+        i += 1
+
+    return db
+
+def normalize_workstream(workstream):
+    return {
+      "id": workstream["id"],
```

Comment:
Idea: `str(workstream["id"])` so that validation would fail if it's not a string. But you could go pretty far with validation, and I think the risk of error is low, so take it or leave it :)

Reply:
Let's leave it :)